src/api/fetchData.js

```diff
@@ -0,0 +1,13 @@
+import { BASE_URL } from '../constants/baseUrl';
+async function fetchData(endpoint, id = 0) {
```

Comment:
here you're creating a generic function to fetchData, but not every api endpoint requires an Id and at that point this will fail.
e.g; Assume you want to fetch a list of restaurants: your api endpoint would be - **/restaurants** (won't be having Id) so this will not work. you will either get a network error or a restaurant with _**Id: 0**_